src/components/NetworkMSBN/index.js

```diff
@@ -43,49 +42,101 @@ import { v4 } from 'uuid';
 import { hasCycles, mergeNetworks } from './helpers';
 
 class NetworkMSBN extends Component {
-  constructor(props) {
-    super(props);
+  state = {
+    connectSubnetwork: null,
+    openSubnetwork: null,
+    firstNodeToConnect: null,
+    editingLinkages: null,
+    addingChildArrow: null,
+  };
 
-    this.state = {
-      connectSubnetwork: null,
-      openSubnetwork: null,
-      firstNodeToConnect: null,
-      editingLinkages: null,
-      addingChildArrow: null,
-    };
+  connecting = false;
 
-    this.connecting = false;
+  componentDidMount() {
+    window.addEventListener('keyup', this.handleKeyup);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keyup', this.handleKeyup);
+  }
+
+  get nodes() {
+    const { linkagesByNode, network, subNetworks } = this.props;
+
+    return subNetworks.map((subNetwork) => {
+      const id = subNetwork.id || subNetwork.name;
+      const linkages = linkagesByNode[subNetwork.id];
```

Comment:
Looks like you can have some troubles here since line above you are declaring `subNetwork.id || subNetwork.name` what seems that `subNetwork.id` can be null, so you also can an undefined or null here, is that right?

Reply:
Yes, there is some network's file that doesn't have the `id` only the `name` so, in these cases, I will treat the name as id.
But in the future, I will check inconsistent flows in this component better. For example, there is a helper.js file with this component and I pretend to remove it or at least, refactor it.

Reply:
👍